integration/simulation/network/geth_utils.go

```diff
@@ -278,148 +294,160 @@ func ConnectTenNetworkBridge(client ethadapter.EthClient, wallets *params.SimWal
-	}
-
-	err = waitForContractDeployment(context.Background(), client, daRegistryContractReceipt.ContractAddress)
-	if err != nil {
-		return nil, nil, fmt.Errorf("DataAvailabilityRegistry contract not available after time. Cause: %w", err)
-	}
-
-	return daRegistryContract, daRegistryContractReceipt, nil
+	return deployProxyContract[NetworkEnclaveRegistry.NetworkEnclaveRegistry](
+		client,
+		contractOwner,
+		bytecode,
+		NetworkEnclaveRegistry.NetworkEnclaveRegistryMetaData,
+		contractOwner.Address(),
+	)
 }
 
-func deployCrossChainContract(client ethadapter.EthClient, contractOwner wallet.Wallet) (*CrossChain.CrossChain, *types.Receipt, error) {
-	bytecode, err := constants.CrossChainBytecode()
+func deployNetworkConfigContract(client ethadapter.EthClient, contractOwner wallet.Wallet, addresses NetworkConfig.NetworkConfigFixedAddresses) (*NetworkConfig.NetworkConfig, *types.Receipt, error) {
+	bytecode, err := constants.NetworkConfigBytecode()
 	if err != nil {
 		return nil, nil, err
 	}
-	crossChainReceipt, err := DeployContract(client, contractOwner, bytecode)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to deploy CrossChain contract from %s. Cause: %w", contractOwner.Address(), err)
-	}
-	crossChainContract, err := CrossChain.NewCrossChain(crossChainReceipt.ContractAddress, client.EthClient())
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to instantiate CrossChain contract. Cause: %w", err)
-	}
+	return deployProxyContract[NetworkConfig.NetworkConfig](
+		client,
+		contractOwner,
+		bytecode,
+		NetworkConfig.NetworkConfigMetaData,
+		addresses,
+		contractOwner.Address(),
+	)
+}
 
-	opts, err := createTransactor(contractOwner)
+func deployDataAvailabilityRegistry(client ethadapter.EthClient, contractOwner wallet.Wallet, messageBus common.Address, enclaveRegistryAddress common.Address) (*DataAvailabilityRegistry.DataAvailabilityRegistry, *types.Receipt, error) {
+	bytecode, err := constants.DataAvailabilityRegistryBytecode()
 	if err != nil {
 		return nil, nil, err
 	}
+	return deployProxyContract[DataAvailabilityRegistry.DataAvailabilityRegistry](
+		client,
+		contractOwner,
+		bytecode,
+		DataAvailabilityRegistry.DataAvailabilityRegistryMetaData,
+		messageBus,
+		enclaveRegistryAddress,
+		contractOwner.Address(),
+	)
+}
 
-	tx, err := crossChainContract.Initialize(opts, contractOwner.Address())
-	if err != nil {
-		return nil, nil, fmt.Errorf("unable to initialize CrossChain contract. Cause: %w", err)
-	}
-
-	_, err = integrationCommon.AwaitReceiptEth(context.Background(), client.EthClient(), tx.Hash(), 25*time.Second)
+func deploMerkleMessageBusContract(client ethadapter.EthClient, contractOwner wallet.Wallet) (*CrossChain.CrossChain, *types.Receipt, error) {
+	bytecode, err := constants.MerkleTreeMessageBusBytecode()
 	if err != nil {
-		return nil, nil, fmt.Errorf("no receipt for CrossChain initialization")
+		return nil, nil, err
 	}
+	return deployProxyContract[CrossChain.CrossChain](
```

Comment:
shouldn't this be merkleTreeMessageBus?

Reply:
good spot. weird that the tests still passed. guess we dont test any of the bridging stuff in the sims?